src/algebra/associated.lean

```diff
@@ -15,6 +16,12 @@ variables {α : Type*} {β : Type*} {γ : Type*} {δ : Type*}
 lemma is_unit.pow [monoid α] {a : α} (n : ℕ) : is_unit a → is_unit (a ^ n) :=
 λ ⟨u, hu⟩, ⟨u ^ n, by simp *⟩
 
+instance invertible_pow {M : Type*} [monoid M] (m : M) [invertible m] (n : ℕ) :
+  invertible (m ^ n) :=
+{ inv_of := ⅟ m ^ n,
+  inv_of_mul_self := by rw [← (commute_inv_of m).symm.mul_pow, inv_of_mul_self, one_pow],
+  mul_inv_of_self := by rw [← (commute_inv_of m).mul_pow, mul_inv_of_self, one_pow] }
+
```

Comment:
These two lemmas probably belong in `algebra.group.power`, but I figured I'd just put them next to each other for now.